charts/simple-app/templates/prometheusrules.yaml

```diff
@@ -0,0 +1,265 @@
+        {{ toYaml . | nindent 8 }}
+        {{- end }}
+    {{- end }}
+
+  - name: {{ .Release.Namespace }}.{{ .Release.Name }}.{{ .Chart.Name }}.ContainerRules
+    rules:
+
+    {{- with .Values.prometheusRules.CPUThrottlingHigh }}
+    - alert: CPUThrottlingHigh
+      annotations:
+        summary: Processes experience elevated CPU throttling.
+        runbook_url: {{ $runbookUrl }}#CPUThrottlingHigh
+        description: >-
+          {{`{{ $value | humanizePercentage }} throttling of CPU in
+          namespace {{ $labels.namespace }} for container {{ $labels.container
+          }} in pod {{ $labels.pod }}.`}}
+      expr: |-
+        sum(
+          increase(
+            container_cpu_cfs_throttled_periods_total{
+              container!="",
+              namespace=~"{{ $targetNamespace }}",
+              pod=~"{{ include "simple-app.fullname" $global }}-.*"
+            }[5m]
+          )
+        ) by (container, pod, namespace)
+
+          /
+
+        sum(
+          increase(
+            container_cpu_cfs_periods_total{}[5m]
+          )
+        ) by (container, pod, namespace)
+
+        > ( {{ .threshold }} / 100 )
+      for: {{ .for }}
+      labels:
+        severity: {{ .severity }}
+        {{- with $values.prometheusRules.additionalRuleLabels }}
+        {{ toYaml . | nindent 8 }}
+        {{- end }}
+    {{- end }}
+
+    {{ with .Values.prometheusRules.ContainerWaiting -}}
+    - alert: ContainerWaiting
+      annotations:
+        summary: Pod container waiting longer than {{ .for }}
+        runbook_url: {{ $runbookUrl }}#containerwaiting
+        description: >-
+          Pod {{`{{`}} $labels.namespace {{`}}`}}/{{`{{`}} $labels.pod {{`}}`}}
+          container {{`{{`}} $labels.container{{`}}`}} has been in waiting
+          state for longer than 1 hour.
+      expr: sum by (namespace, pod, container) (kube_pod_container_status_waiting_reason{job="kube-state-metrics", namespace=~"{{ $targetNamespace }}"}) > 0
+      for: {{ .for }}
+      labels:
+        severity: {{ .severity }}
+        {{- with $values.prometheusRules.additionalRuleLabels }}
+        {{ toYaml . | nindent 8 }}
+        {{- end }}
+    {{- end }}
+
+  - name: {{ .Release.Namespace }}.{{ .Release.Name }}.{{ .Chart.Name }}.DeploymentRules
+    rules:
+
+    {{ with .Values.prometheusRules.DeploymentGenerationMismatch -}}
+    - alert: DeploymentGenerationMismatch
+      annotations:
+        summary: Deployment generation mismatch due to possible roll-back
+        runbook_url: {{ $runbookUrl }}#deploymentgenerationmismatch
+        description: >-
+          Deployment generation for {{`{{`}} $labels.namespace
+          {{`}}`}}/{{`{{`}} $labels.deployment {{`}}`}} does not match, this
+          indicates that the Deployment has failed but has not been rolled
+          back.
+      expr: |-
+        kube_deployment_status_observed_generation{job="kube-state-metrics", namespace=~"{{ $targetNamespace }}"}
```

Comment:
should have a filter on `deployment="{{ include "simple-app.fullname" . }}"`

Reply:
Good catch... fixing.